### ISSUE TITLE
Enhancement: [consistent-type-assertions] Prefer type annotation over assertion for arrays

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6374]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Code should not use an `as` type assertion when a `:` type annotation would do. In some cases (see examples later) they act the same, but `as` assertions are generally something that should be avoided whenever possible. `as` removes some type safety from TypeScript. Not ideal.

### Fail Cases

```typescript
const errorMessages = [] as string[];
```

### Pass Cases

```typescript
const errorMessages: string[] = [];
```

### Additional Info

This should also trigger for arrays that have contents in them.

```diff
- const errorMessages = ['a'] as string[];
+ const errorMessages: string[] = ['a'];
```

Alas, knowing whether the assertion actually was doing anything important is blocked on my favorite TypeScript issue https://github.com/microsoft/TypeScript/issues/9879. But since folks really shouldn't be using an assertion for a variable with an initial value, I'm ok with this edge case bug being in the rule:

```diff
// The + is now an error... 😈 deal with it!
- const errorMessages2 = [Math.random() ? 'a' : 'b'] as 'a'[];
+ const errorMessages3: 'a'[] = [Math.random() ? 'a' : 'b'];
```

**Comment by bradzacher** [comment]
`satisfies` should do the right thing here too I believe.

I think that `consistent-type-assertions` also would cover this with its assertion style option - but perhaps it needs a specific option like `objectLiteralTypeAssertions` but for arrays?

**Comment by JoshuaKGoldberg** [comment]
Good point! Switching this to a feature enhancement request.

**Comment by danvk** [comment]
Just so I understand… is the proposal here to add a new `arrayLiteralTypeAssertions` option for `consistent-type-assertions`, analogous to `objectLiteralTypeAssertions`? Would it be OK to extend `objectLiteralTypeAssertions` to cover arrays since, after all, `typeof [] === 'object'`?

**Comment by Josh-Cena** [comment]
`objectLiteralTypeAssertions` is about _object literals_, which means a literal with `{}`. It would be quite weird to conflate array literals with them.

**Comment by danvk** [comment]
I've put up a PR for this, #6749.
